dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1773,4 +1826,668 @@ public void getDirectEpersonGroups() throws Exception {
+        String tokenForEPerson = registrationDataService.findByEmail(context, ePerson.getEmail()).getToken();
+        String token = getAuthToken(admin.getEmail(), password);
+        PasswordHash oldPassword = ePersonService.getPasswordHash(ePerson);
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON)
+                                     .param("token", tokenForEPerson))
+                        .andExpect(status().isOk());
+
+        PasswordHash newPasswordHash = ePersonService.getPasswordHash(ePerson);
+        assertFalse(oldPassword.equals(newPasswordHash));
+        assertTrue(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.deleteByToken(context, tokenForEPerson);
+        context.restoreAuthSystemState();
+    }
+
+
+    @Test
+    public void patchReplacePasswordWithRandomTokenPatchFail() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .withPassword(password)
+                                        .build();
+
+        String newPassword = "newpassword";
+
+        context.restoreAuthSystemState();
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/password", newPassword);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+        accountService.sendRegistrationInfo(context, ePerson.getEmail());
+        String tokenForEPerson = registrationDataService.findByEmail(context, ePerson.getEmail()).getToken();
+        String token = getAuthToken(admin.getEmail(), password);
+        PasswordHash oldPassword = ePersonService.getPasswordHash(ePerson);
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON)
+                                     .param("token", "RandomToken"))
+                        .andExpect(status().isForbidden());
+
+        PasswordHash newPasswordHash = ePersonService.getPasswordHash(ePerson);
+        assertTrue(StringUtils.equalsIgnoreCase(oldPassword.getHashString(),newPasswordHash.getHashString()));
+        assertFalse(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
+        assertTrue(StringUtils.equals(registrationDataService.findByEmail(context, ePerson.getEmail()).getToken(),
```

Comment:
I'm finding these asserts hard to read/understand.  I'd recommend these be `assertEquals()`, `assertNotNull` and `assertEquals()`

Reply:
This has been adjusted